### ISSUE TITLE
Consider to speed up the generating process

### ISSUE THREAD

**Issue by Liam0205** [theme-next/hexo-theme-next#661]
### I agree and want to create new issue 

- [X] Yes, I was on [Hexo Docs page](https://hexo.io/docs/), especially on [Templates](https://hexo.io/docs/templates.html), [Variables](https://hexo.io/docs/variables.html), [Helpers](https://hexo.io/docs/helpers.html) and [Troubleshooting](https://hexo.io/docs/troubleshooting.html).
- [X] Yes, I was on [NexT Documentation Site](http://theme-next.org/docs/).
- [X] And yes, I already searched for current [issues](https://github.com/theme-next/hexo-theme-next/issues?utf8=%E2%9C%93&q=is%3Aissue) and this is not help to me.

***

### Expected behavior 

When I was using old version of NexT (0.x), generating 250+ posts costs around 1.4 minutes. I consider this is the expected behavior of Hexo generating speed.

### Actual behavior 

Now, generating 280 posts cost more than 7 minutes, and one day it costs me 20 minutes, which is not considered to be tolerated.

### Steps to reproduce the behavior 

```bash
$ hexo clean && time hexo generate
```

* Link to demo site with this feature: N/A
* Link(s) to source code or any usefull link(s): N/A

### What I've tried

Here are what I've tried:

* specify programming language for code blocks in all posts, based on [this post](https://blog.xinshangshangxin.com/2015/06/29/hexo3%E4%B8%ADhexo-g%E7%94%9F%E6%88%90%E9%80%9F%E5%BA%A6%E8%BF%87%E6%85%A2/) and [this post](https://www.hahack.com/codes/hexo-3-speed-up/).
* enable NexT's cache, introduced in version 6.0.0, based on https://github.com/theme-next/hexo-theme-next/issues/539#issuecomment-449922610.

**NexT Scheme:**

- [x] All schemes
- [ ] Muse
- [ ] Mist
- [ ] Pisces
- [ ] Gemini

Hexo's and NexT's version number, configuration of Hexo and NexT, and environment information could be found in #658 .

**Comment by Liam0205** [comment]
Test for cache-disabled. [disable_cache.log](https://github.com/theme-next/hexo-theme-next/files/2939326/disable_cache.log)


```bash
$ hexo clean && time hexo generate > disable_cache.log 2>&1
INFO  Deleted database.
INFO  Deleted public folder.

real    10m40.752s
user    0m0.030s
sys     0m0.291s
```

Test for cache-enabled. (didn't get the announced 3x speed-up.)
[enable_cache.log](https://github.com/theme-next/hexo-theme-next/files/2939350/enable_cache.log)


```bash
$ hexo clean && time hexo generate > enable_cache.log 2>&1
INFO  Deleted database.
INFO  Deleted public folder.

real    7m58.032s
user    0m0.031s
sys     0m0.306s
```

**Comment by JiangTJ** [comment]
Same issue in all schemes.
See https://hexo.io/docs/templates.html#Optimization.
Fragment Caching may be a good way, but it takes a lot of work.

**Comment by Liam0205** [comment]
> Same issue in all schemes.

Scheme or theme? Is this a NexT specific problem?

> See https://hexo.io/docs/templates.html#Optimization.
> Fragment Caching may be a good way, but it takes a lot of work.

Do your team has a plan for this?

**Comment by ivan-nginx** [comment]
@Liam0205 issue will be fixed soon. I've already turnkey solution, need just to debug it.

Releated to #539.

**Comment by Liam0205** [comment]
@ivan-nginx Glad to know!

If there is any progress of it, or if any test task needed, please let me know.

**Comment by JiangTJ** [comment]
I think use `partial` helper instead of `include` swig tag may be improved to some extent.

`partial`  read data from `theme.views` object.
https://github.com/hexojs/hexo/blob/a6dc0ea28dddad1b5f1bad7c6f86f1e0627b564a/lib/plugins/helper/partial.js#L12

`include` read from disk file.
![image](https://user-images.githubusercontent.com/15902347/56210651-e24c9800-6088-11e9-97f8-98eae2ce9539.png)

**Comment by jiangtj** [comment]
https://github.com/theme-next/hexo-theme-next/pull/969  I don't have any ideas now. . .

**Comment by stevenjoezhang** [comment]
So... how do
https://github.com/hexojs/hexo-generator-tag
and
https://github.com/hexojs/hexo-generator-category
deal with `_draft` 😥

#42 #87

**Comment by jiangtj** [comment]
https://github.com/hexojs/hexo/blob/b6394c78751d2be81c3fd7c243fca37c552cd83a/lib/plugins/helper/tagcloud.js#L196

Same, filter. We only used it in one place....

**Comment by jiangtj** [comment]
https://github.com/hexojs/hexo-theme-landscape/blob/7e0bd67ddf5d41d161437e2343da5d45c076a072/layout/layout.ejs#L11

Cache...

**Comment by Liam0205** [comment]
After #971 being merged, the generating speed is much faster than before.

My great appreciation would be shown to @jiangtj and the team for the continuous efforts on this issue.
